bn-apps/ledger-sync/ledger-sync-service/src/main/kotlin/com/r3/businessnetworks/ledgersync/RequestLedgersSyncFlow.kt

```diff
@@ -52,11 +52,13 @@ class RespondLedgerSyncFlow(
 
     @Suspendable
     override fun call() {
-        val theirs = otherSideSession.receive<List<SecureHash>>().unwrap { it }
+        val (theirs, their_VR) = otherSideSession.receive<Pair<List<SecureHash>, List<SecureHash>>>().unwrap { it }
```

Comment:
why do we need to receive both of the lists? Can we not merge those `before` sending? I believe that would also reduce traffic significantly.

Reply:
cause both sides need to exclude the recycled_tx from both sides. if node1 sends (ours - ours_vr) to node2, node2 has no way to exclude the recycled_tx at its end.

Reply:
So let's say if `Node A` has gotten txs A, B, C live on the ledger while D, E has been recycled. 
What you are proposing is to send [(ABC), (D,E)] as two separate lists. Why can't we just bundle all of them together? I.e. send (A, B, C, D, E) as if D and E have not been removed at all? Then `Node B` would just assume that D and E are present at `Node A's` side and wouldn't attempt to recover them.

Reply:
in this case, if Node B has txs A, B, C and lost D, E (no recycled transaction), Node B will be considered missing txs D and E

Reply:
Yes, which is how it should work right? Even though D and E are recyclable, node A would still want to recover them as maybe it doesn't want to use VR at all and would like to keep everything on the ledger.

Reply:
D, E are RECYCLED tx not just RECYCLABLE tx, meaning D, E exists in Recyclable_tx but not in node_tx. So actually Node B cannot recover them from Node A.

Reply:
plus recycled tx cannot be filtered by participants (has been cleaned up), thus the recycled tx list here is all recycled tx, including those not exactly have NodeA and NodeB as participants.

Reply:
Ok let's assume:

Node 1
node_tx    recycled_tx
A              F
B
C

Node 2     
node_tx    recycled_tx
A              C
B              D
F              E
G

1. Node 1 sends (A, B, C) to the Node 2
2. Node 2 sees that A, B are present at theirs vault and that C has been recycled, so they aren't interested in recovering anything
3. Node 2 thinks that Node 1 is missing F, G. Node 2 sends the following reply [(), (F, G)].
4. Node 1 receives reply from Node 2. Node 1 iterates through (F, G). Node 1 doesn't attempt to recover F, as it exists in its local recycled table. Node 1 recovers only G.

Reply:
If I am not wrong, in this scenario, the exclusion of recycled_tx is done at both sides. Node 2 as receiver will exclude its own recycled_tx in LedgerSyncFindings first and after sending LederSyncFindings back to Node1, Node1 will exclude its recycled_tx again. The difference between our approach is just excluding recycled_tx altogether or separately at each node. Is it because there is some privacy concern for sending recycled_tx list?

Reply:
Nopes, the main points are:
1. Not to break an API if this is not massively needed
2. Send less data over the wire
3. Let the nodes drive what they would like to recover rather than pushing this decision onto their counterparts

Reply:
okie, new commit made.